src/pages/Exchange/index.tsx

```diff
@@ -244,135 +279,155 @@ const Exchange: React.FC<ExchangeProps> = ({
   };
 
   return (
-    <IonPage id="exchange-page">
-      <Loader showLoading={isLoading} delay={0} />
-      {assetSent && assetReceived && markets.length > 0 && (
-        <PinModal
-          open={modalOpen}
-          title="Unlock your seed"
-          description={`Enter your secret PIN to send ${sentAmount} ${
-            isLbtc(assetSent.asset, network) ? lbtcUnit : assetSent.ticker
-          } and receive ${receivedAmount} ${isLbtc(assetReceived.asset, network) ? lbtcUnit : assetReceived.ticker}.`}
-          onConfirm={onPinConfirm}
-          onClose={() => {
-            setModalOpen(false);
-          }}
-          isWrongPin={isWrongPin}
-          needReset={needReset}
-          setNeedReset={setNeedReset}
-          setIsWrongPin={setIsWrongPin}
-        />
-      )}
-
-      {assetSent && markets.length > 0 && (
-        <IonContent className="exchange-content">
-          <Refresher />
-          <IonGrid className="ion-no-padding ion-padding-top">
-            <Header
-              hasBackButton={false}
-              hasCloseButton={true}
-              customRightButton={
-                <IonButton className="custom-right-button" onClick={() => history.push('/history')}>
-                  <img src={tradeHistory} alt="trade history" />
-                </IonButton>
+    <>
+      <IonPage id="exchange-page">
+        <Loader showLoading={isBusyMakingTrade} delay={0} />
+        {/* Check location because screens are always loaded on Ionic */}
+        {/* Otherwise, loaders will show up on the other screens */}
+        {history.location.pathname === '/exchange' && (
+          <Loader
+            showLoading={isFetchingMarkets}
+            message="Discovering TDEX providers with best liquidity..."
+            delay={0}
+            backdropDismiss={true}
+            duration={15000}
+            onDidDismiss={() => {
+              if (markets.length === 0 || bestTrade === undefined) {
```

Comment:
not sure if markets.length (no markets) is the same error of bestTrade === undefined (discovery has failed). I would show different errors message for bestTrade undefined case, what's your opinion?

Reply:
Here an alert box is shown when Loader is dismissed, with { header: 'No providers available', message: 'Liquidity providers on Tor network can take a long time to respond or all your providers are offline.'}
I would say that it encompass both cases `markets.length === 0` and `bestTrade === undefined`.

Reply:
So imo "Discovering TDEX providers with best liquidity..." and 'No providers available' are complementary